### PR TITLE
Using a welder to repair a mining drone now follows standard behaviour

### LINKED ISSUE #23019: You can repair a minebot with a welding tool and not take eye damage
It was an upgraded industrial welding tool, but I think they are just as eye damaging as normal welders.

There was also no click cooldown

### PR DESCRIPTION
**Pull request by Mervill**
:cl: Mervill
fix: Using a welder to repair a mining drone now follows standard behaviour
fix: Redeeming the mining voucher for a mining drone now also provides welding goggles
/:cl:

Note that this also fixes an undiscovered bug where _repairing the mining drone was free_.

fixes #23019

### CODE REVIEW

**Comment by XDTM** [comment]
Check mech code, i'm pretty sure they have the same issue.

File: code/modules/mining/minebot.dm [hunk 1]
```diff
@@ -75,8 +75,9 @@
 			if(maxHealth == health)
 				user << "<span class='info'>[src] is at full integrity.</span>"
 			else
-				adjustBruteLoss(-10)
-				user << "<span class='info'>You repair some of the armor on [src].</span>"
+				if(W.remove_fuel(1, user))
+					adjustBruteLoss(-10)
+					user << "<span class='info'>You repair some of the armor on [src].</span>"
```
Comment (Cruix) [inline code/modules/mining/minebot.dm]: 1 unit of fuel is removed in /obj/item/weapon/weldingtool/afterattack(), return 1 here to avoid doubling the cost.

**Comment by Cruix** [comment]
This will also flash miners eyes and force them to get a welding mask or other eye protection to repair their bot without going blind.

File: code/modules/mining/minebot.dm (hunk 1 above)
Comment (Mervill) [inline code/modules/mining/minebot.dm]: I suspected something like that was happening since remove_fuel uses 0 as it's first argument in a bunch of places

**Comment by YakumoChen** [comment]
Yeah this basically makes it a massive pain in the ass unless minebot now comes with free welding mask / goggles

**Comment by YakumoChen** [comment]
doubly so since most miners are used to wearing mesons and never, ever, ever taking them off.

**Comment by Mervill** [comment]
Lavaland is @KorPhaeron's thing, lets see what he thinks

**Comment by KorPhaeron** [comment]
Ideally it'd not eyeflash, but that is probably garbage code, so giving them a pair of welding goggles free when they buy a mining drone probably makes the most sense

**Pull request merged**